packages/combobox/src/Combobox.spec.tsx

```diff
@@ -1288,27 +1288,52 @@ describe('packages/combobox', () => {
     /**
      * onClear
      */
-    test('Clear button calls onClear callback', () => {
-      const initialValue = select === 'multiple' ? ['apple'] : 'apple';
-      const onClear = jest.fn();
-      const { clearButtonEl } = renderCombobox(select, {
-        initialValue,
-        onClear,
-      });
-      userEvent.click(clearButtonEl!);
-      expect(onClear).toHaveBeenCalled();
-    });
+    describe('onClear', () => {
+      test('Clear button calls onClear callback', () => {
+        const initialValue = select === 'multiple' ? ['apple'] : 'apple';
+        const onClear = jest.fn();
+        const { clearButtonEl } = renderCombobox(select, {
+          initialValue,
+          onClear,
+        });
+        userEvent.click(clearButtonEl!);
+        expect(onClear).toHaveBeenCalled();
+      });
+
+      test('Clear button does not force the menu to reopen', () => {
+        const initialValue = select === 'multiple' ? ['apple'] : 'apple';
+        const onClear = jest.fn();
+        const { clearButtonEl, queryByRole } = renderCombobox(select, {
+          initialValue,
+          onClear,
+        });
+        expect(queryByRole('listbox')).not.toBeInTheDocument();
+        userEvent.click(clearButtonEl!);
+        expect(queryByRole('listbox')).not.toBeInTheDocument();
+      });
+
+      test('Clear button clears the value of the input', () => {
+        const value = 'apple';
+        const { inputEl, clearButtonEl } = renderCombobox(select, {
+          value,
+        });
+
+        expect(inputEl).toHaveValue('Apple');
+        userEvent.click(clearButtonEl!);
+        expect(inputEl).toHaveValue('');
```

Comment:
I don't think this test should be passing. If you provide `value` (i.e. controlled component), then the clear button shouldn't do anything until you update the value prop externally

Reply:
Updated the test